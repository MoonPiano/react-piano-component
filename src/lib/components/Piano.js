export default function Piano() {
  return null;
}
