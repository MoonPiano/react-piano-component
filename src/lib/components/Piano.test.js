import React from 'react';
import { mount } from 'enzyme';
import Piano from './Piano';

describe('Piano', () => {
  it('calls renderPianoKey() with the expected props', () => {
    const renderPianoKey = jest.fn();
    mount(<Piano startNote={'C1'}
      endNote={'C2'}
      renderPianoKey={renderPianoKey}
    />);

    expect(renderPianoKey.mock.calls).toMatchSnapshot();
  });

  it('renders a custom IntrumentAudio', () => {
    const CustomAudio = () => null;
    const renderPianoKey = jest.fn();

    const wrapper = mount(<Piano startNote={'C1'}
      endNote={'C2'}
      CustomAudio={CustomAudio}
      renderPianoKey={renderPianoKey}
    />);

    expect(wrapper.find(CustomAudio).exists()).toBe(true);
  });
});
