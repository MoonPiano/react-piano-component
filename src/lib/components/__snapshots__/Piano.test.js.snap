// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Piano calls renderPianoKey() with the expected props 1`] = `
Array [
  Array [
    Object {
      "isNoteAccidental": false,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "C1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": true,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "C#1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": false,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "D1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": true,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "D#1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": false,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "E1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": false,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "F1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": true,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "F#1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": false,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "G1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": true,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "G#1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": false,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "A1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": true,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "A#1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": false,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "B1",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
  Array [
    Object {
      "isNoteAccidental": false,
      "isNotePlaying": false,
      "keyboardShortcuts": Array [],
      "note": "C2",
      "startPlayingNote": [Function],
      "stopPlayingNote": [Function],
    },
  ],
]
`;
